### DISCUSSION TITLE
Upgrading openssl from 1.1.1 to openssl 3.0.9 causes problems with application services

### DISCUSSION THREAD

**Discussion by Wormhole-design** [openssl/openssl#22608] [Q&A]
**Background：** When we try to upgrade openssl from 1.1.1 to openssl3.0.9, the LDAP application service cannot be used normally.
**Version description：**
- LDAP client： linux openssl3.0.9
- LDAP server（AD domain）：windows2008 openssl1.0.2 

**Details of the problem：**
As an LDAP client, our backend will initiate a request to the LDAP server and establish a connection. We found through packet capture that after our backend sends Client Hello to the LDAP server, the LDAP server will generate an RST response to let the client to reset the link.

**Comment by t8m** [comment]
Unfortunately there is too little information here to be able to help you. Perhaps the server does not accept newer TLS protocol versions? You have to use non-default seclevel in ciphers setting with 3.x version (i.e. use something like `DEFAULT:@SECLEVEL=0` for cipher configuration string) to be able to communicate with TLS-1.0 or 1.1. Also I'd suggest trying to change the maximum protocol version to TLS-1.2 or even lower.

**Reply by Wormhole-design** [reply]
We have captured and analyzed the packets and found that the communication between them is TLS v1.2. We compared the differences before and after the openssl upgrade and found that after the upgrade, the request did not seem to go to the AD domain application because there was no relevant log printing in the AD domain. We suspect that it is caused by the incompatibility between windows2008 openssl1.0.2 and linux openssl 3.0.9 communication?

**Reply by Wormhole-design** [reply]
We used linux openLDAP as the server and found that normal communication was possible; we also used linux freeIPA as the server and found that normal communication was also possible. The openssl of these two servers is also 1.0.2.

**Reply by t8m** [reply]
As I said the problem can possibly be that the 3.0 client tries to negotiate TLS-1.3 and/or it might be related to demoting some algorithms to SECLEVEL 0 in 3.0 in particular SHA1 or that something else has changed in the default ciphers used. I can only recommend trying to change the ciphers used by the 3.0 client. If it is not possible I am afraid there is not much else that could help you with.
